src/py42/services/_connection.py

```diff
@@ -195,6 +195,8 @@ def request(
                 # setting this manually speeds up read times
                 response.encoding = u"utf-8"
 
+            debug.logger.debug(u"Response\n {0}".format(str(response)))
```

Comment:
Can this be like this instead? No need for a newline, and gives the status and text of response. 
```
debug.logger.debug(u"Response: status={0}, data={1}".format(response.status_code, response.text))
```

Reply:
I suppose we should also first check if the response is streamed and not use `.text` in that case, so it doesn't block and consume the stream just to log.